tbx/navigation/models.py

```diff
@@ -41,22 +39,18 @@ class NavigationSettings(BaseSiteSetting, ClusterableModel):
     def save(self, **kwargs):
         super().save(**kwargs)
 
-        fragment_keys = ["primarynav"]
+        fragment_keys = ["primarynav", "primarynavmobile", "footerlinks"]
 
         # The fragment cache varies on:
-        # the current site pk, whether used in preview, or in the pattern library
+        # the current site pk, whether used in the pattern library
 
-        request_is_preview_options = [True, False]
         # NOTE: `is_pattern_library` returns True if pattern is being rendered in the pattern library,
         # but it doesn't return False if otherwise, hence the empty string instead of False
         is_pattern_library_options = [True, ""]
 
-        # Generate all combinations of `request_is_preview` and `is_pattern_library`
-        combinations = product(request_is_preview_options, is_pattern_library_options)
-
         keys = [
-            make_template_fragment_key(key, vary_on=(self.site.pk,) + combination)
+            make_template_fragment_key(key, vary_on=(self.site.pk, is_pattern_library))
```

Comment:
Andy commented that this felt quite verbose and wondered if a more simple solution was to just use a `last_updated` timestamp on the model, although:
- that would need adding to all the page models as we don't inherit from a base page
- I am not sure that this covers the pattern library case

Reply:
I didn't want to deviate massively from the logic that was already there. As `wagtailcache` is preview-aware we only need the site and pattern library, which already simplifies the combinations